src/monitor.js

```diff
@@ -33,14 +33,23 @@ class Monitor {
       return sentry;
     }).catch(err => {});
 
-    this._sentry.then(sentry => {
+    return this._sentry.then(sentry => {
       sentry.client.captureException(err, {
         tags: _.defaults({
           prefix: this._opts.project + (this._opts.prefix || '.root'),
           process: this._opts.process || 'unknown',
         }, tags),
         level,
       });
+
+      return new Promise((accept, reject) => {
+        sentry.client.once('logged', () => {
+          accept('Succesfully logged error to Sentry.');
```

Comment:
No reason to send an argument... Usually:
```js
return new Promise((accept, reject) => {
  sentry.client.once('logged', accept);
  sentry.client.once('error', reject);
});
```
But even this leaks... Sorry...

We probably have to do:
```js
return new Promise((accept, reject) => {
  let onLogged, onError;
  onLogged = () => {
    sentry.client.removeListener('error', onError);
    accept();
  };
  onError = (err) => {
    sentry.client.removeListener('logged', onLogged);
    reject(err);
  }
  sentry.client.once('logged', onLogged);
  sentry.client.once('error', onError);
});
```
Yes, this sucks... But it doesn't leak memory... hopefully this doesn't matter... but if one day we have 100k internal errors, it would suck if we also got a memory consumption issue :)